src/dune_rules/lock_dir.ml

```diff
@@ -147,12 +147,16 @@ let dev_tool_to_path_segment dev_tool =
   dev_tool |> Dev_tool.package_name |> Package_name.to_string |> Path.Local.of_string
 ;;
 
-let dev_tool_source_lock_dir dev_tool =
-  let dev_tools_path = Path.Source.(relative root "dev-tools.locks") in
+(* This function returns the lock dir that is created outside the build system. *)
+let dev_tool_untracked_lock_dir dev_tool =
+  let dev_tools_path = Path.Build.relative Path.Build.root ".dev-tools.locks" in
```

Comment:
Although this directory lives in the _build directory, the Path.Build.t isn't ideal for it because it means that it cannot be watched. How about trying Path.External.t instead?